src/api/baseapi.cpp

```diff
@@ -1889,15 +1893,19 @@ void TessBaseAPI::End() {
     delete paragraph_models_;
     paragraph_models_ = nullptr;
   }
+#ifndef DISABLED_LEGACY_ENGINE
   if (osd_tesseract_ == tesseract_) {
     osd_tesseract_ = nullptr;
   }
+#endif // ndef DISABLED_LEGACY_ENGINE
   delete tesseract_;
   tesseract_ = nullptr;
+#ifndef DISABLED_LEGACY_ENGINE
   delete osd_tesseract_;
   osd_tesseract_ = nullptr;
   delete equ_detect_;
   equ_detect_ = nullptr;
+#endif // ndef DISABLED_LEGACY_ENGINE
```

Comment:
That lines can be moved up into the previous conditional block.

Reply:
Done.